### PR TITLE
Adding Support for accept and accept4

### LINKED ISSUE #284: Support for `accept{,4}`
With #282 in place, it should be now possible to implement `accept` and `accept4`.

### PR DESCRIPTION
**Pull request by Deepansharora27**
I have made a draft PR for this and have made some inital changes

Fixes #284

### CODE REVIEW

File: libc-bottom-half/headers/public/__header_sys_socket.h [hunk 1]
```diff
@@ -34,6 +34,9 @@
 extern "C" {
 #endif
 
+int accept(int socket, struct sockaddr *restrict addr, socklen_t *restrict addrlen);
+int accept4(int socket, struct sockaddr *restrict addr, socklen_t *restrict addrlen, int flags);
+
```
Comment (sbc100) [inline libc-bottom-half/headers/public/__header_sys_socket.h:39]: Why do these need to be declared here, but not other socket APIs?   

It looks like `top-half/musl/include/sys/socket.h` can be modified instead to declare these maybe?

Comment (Deepansharora27) [reply libc-bottom-half/headers/public/__header_sys_socket.h:39]: Sure I will do so

Comment (sunfishcode) [reply libc-bottom-half/headers/public/__header_sys_socket.h:39]: Yes.

Comment (Deepansharora27) [reply libc-bottom-half/headers/public/__header_sys_socket.h:39]: Made the Change

**Comment by haraldh** [comment]
move to https://github.com/WebAssembly/wasi-libc/pull/287

**Pull request closed without merging**